devito/visitors.py

```diff
@@ -162,6 +163,11 @@ def visit_tuple(self, o, ret=None, queue=None):
             ret = self.visit(i, ret=ret, queue=queue)
         return ret
 
+    def visit_Node(self, o, ret=None, queue=None):
+        for i in o.children:
+            ret = self.visit(i, ret=ret, queue=queue)
+        return ret
```

Comment:
This looks odd to me. Don't we need to perform some kind of reduction here over `ret`, or is the return value negligible?